socorro/processor/processor_app.py

```diff
@@ -77,6 +77,43 @@ def get_application_defaults():
             "destination.crashstorage_class": FSDatedPermanentStorage,
         }
 
+    def _capture_error(self, crash_id, exc_type, exc_value, exc_tb):
+        """Capture an error in sentry if able
+
+        The `exc_*` arguments come from calling `sys.exc_info`.
+
+        :arg crash_id: a crash id
+        :arg exc_type: the exc class
+        :arg exc_value: the exception
+        :arg exc_tb: the traceback for the exception
+
+        """
+
+        if self.config.sentry and self.config.sentry.dsn:
+            try:
+                if isinstance(exc_value, collections.Sequence):
+                    # Then it's already an iterable!
+                    exceptions = exc_value
+                else:
+                    exceptions = [exc_value]
+                client = raven_client.get_client(self.config.sentry.dsn)
+                client.context.activate()
+                client.context.merge({'extra': {
+                    'crash_id': crash_id,
+                }})
+                try:
+                    for exception in exceptions:
+                        identifier = client.captureException(exception)
+                        self.config.logger.info(
+                            'Error captured in Sentry! Reference: {}'.format(identifier)
+                        )
+                finally:
+                    client.context.clear()
+            except Exception:
+                self.config.logger.error('Unable to report error with Raven', exc_info=True)
+        else:
+            self.config.logger.warning('Sentry DSN is not configured and an exception happened')
+
     def _transform(self, crash_id):
```

Comment:
At some point, we might want to refactor `_transform` so that it's more coherent about error handling through all stages of transforming without nearly as many try/except and other blocks. I bet we could unify it and make it much easier to follow.